src/session/session_dhandle.c

```diff
@@ -640,6 +653,9 @@ __wt_session_get_btree_ckpt(WT_SESSION_IMPL *session, const char *uri, const cha
 
     } while (is_unnamed_ckpt && (ret == WT_NOTFOUND || ret == EBUSY));
 
+err:
+    __wt_free(session, checkpoint);
+    __wt_free(session, hs_checkpoint);
```

Comment:
While this is correct, it also took some thought about why this was added when no change used those two things. It might be clearer with:
if (ret != 0) {
   __wt_free(...)
...
}

I don't have a strong opinion.

Reply:
Even if `ret` is `0`, we should make sure things get cleaned up though. This is true that this label comes from my changes but I believe it is good to free the allocated memory in all cases if the allocated memory should not go beyond the scope of the function.